app/api/events.py

```diff
@@ -857,3 +857,45 @@ def clear_export_urls(event):
     event.xcal_url = None
     event.pentabarf_url = None
     save_to_db(event)
+
+
+class UpcomingEventList(EventList):
+    """
+    List Upcoming Events
+    """
+
+    def before_get(self, args, kwargs):
+        """
+        method for assigning schema based on admin access
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        if 'Authorization' in request.headers and (
+            has_access('is_admin') or kwargs.get('user_id')
+        ):
+            self.schema = EventSchema
+        else:
+            self.schema = EventSchemaPublic
+
+        self.schema.self_view_many = 'v1.upcoming_event_list'
+
+    def query(self, view_kwargs):
+        """
+        query method for upcoming events list
+        :param view_kwargs:
+        :return:
+        """
+        current_time = datetime.now(pytz.utc)
+        query_ = self.session.query(Event).filter(
+            Event.ends_at > current_time,
+            Event.state=='published',
+            Event.privacy=='public').order_by(Event.starts_at)
+        return query_
+
+    schema = EventSchema
+    data_layer = {
+        'session': db.session,
+        'model': Event,
+        'methods': {'query': query},
+    }
```

Comment:
This is already defined in superclass as well I believe

Reply:
The query method is not invoked for returning the desired events without defining data_layer again in the child class. It should take methods in data_layer variable from superclass, but this isn't happening.

Reply:
Hmm. The library is not really well maintained. OK

Reply:
Check other things. If they are working or not

Reply:
Yeah, other things are working fine.